Source/ParleyTests/MessagesManagerTests.swift

```diff
@@ -0,0 +1,274 @@
+import XCTest
+@testable import Parley
+
+final internal class MessagesManagerTests: XCTestCase {
+    
+    struct MessagesManagerTestsError: Error {
+        let message: String
+    }
+    
+    private let MESSAGE_WELCOME_TEXT = "Welcome message";
+    private let MESSAGE_STICKY_TEXT = "Sticky message";
+    
+    private var messagesManager = MessagesManager()
+    private var dataSource: ParleyDataSource? = ParleyInMemoryDataSource()
+    
+    override func setUp() {
+        messagesManager = MessagesManager()
+        dataSource = ParleyInMemoryDataSource()
+        Parley.shared.dataSource = dataSource
+     }
+    
+    func testMessagesManager_ShouldBeEmpty_WhenCreated() {
+        XCTAssertTrue(messagesManager.messages.isEmpty)
+        XCTAssertNil(messagesManager.welcomeMessage)
+        XCTAssertNil(messagesManager.stickyMessage)
+        XCTAssertNil(messagesManager.paging)
+        XCTAssertNil(messagesManager.lastSentMessage)
+        XCTAssertTrue(messagesManager.pendingMessages.isEmpty)
+        XCTAssertNil(messagesManager.getOldestMessage())
+    }
+    
+    func testMessageManager_ShouldBeCorrectlyConfigured_WhenStartingTheMessagesMannagerWithAMessageCollection() {
```

Comment:
In the `when` we already know it's about the messages manager, therefore, we can shorten the name:

Something like: 
`testMessageManager_ShouldBeCorrectlyConfigured_WhenStartingWithAMessageCollection`